docs/contributing/commit-discipline.md

```diff
@@ -86,87 +86,224 @@ check out GitHub's excellent [blog post](https://github.blog/2022-06-30-write-be
-  the "partially" and close the issue. `Fixes part of #1234` is a good alternative.
-- Any paragraph content in the commit message should be line-wrapped
-  to about 68 characters per line, but no more than 70, so that your
-  commit message will be reasonably readable in `git log` in a normal
-  terminal. You may find it helpful to:
-  - configure Git to use your preferred editor, with the EDITOR
-    environment variable or `git config --global core.editor`, and
-  - configure the editor to automatically wrap text to 70 or fewer
-    columns per line (all text editors support this).
+### Commit summary, part 1
+
+The first part of the commit summary should only be 1-2 **lower-case**
+words, followed by a `:`, describing what the part of the product the
+commit changes. These prefixes are essential for maintainers to
+efficiently skim commits when doing release management or
+investigating regressions.
+
+Common examples include: settings, message feed, compose, left
+sidebar, right sidebar, recent (for **Recent conversations**), search,
+markdown, tooltips, popovers, drafts, integrations, email, docs, help,
+and api docs; issue trackers can be a good source of inspiration (but
+may be overly verbose).
+
+When it's possible to do so concisely, it's helpful to be a little more
+specific, e.g., emoji, spoilers, polls. However, a simple `settings:` is better
+than a lengthy description of a specific setting.
+
+If your commit doesn't cleanly map to a part of the product, you might
+use something like "css" for CSS-only changes, or the name of the file
+or technical subsystem principally being modified (not the full path,
+so `realm_icon`, not `zerver/lib/realm_icon.py`).
+
+There is no need to be creative here! If one of the examples above
+fits your commit, use it. Consistency makes it easier for others to
+scan commit messages to find what they need.
+
+Additional tips:
+
+- Use lowercase (e.g., "settings", not "Settings").
+- If it's hard to find a 1-2 word description of the part of the code base
+  affected by your commit, consider again whether you have structured your
+  commits well.
+- Never use a generic term like "bug", "fix", or "refactor".
+
+### Commit summary, part 2
+
+This is a **complete sentence** that briefly summarizes your changes. There are
+a few rules to keep in mind:
+
+- Start the sentence with an
+  [imperative](https://en.wikipedia.org/wiki/Imperative_mood) verb, e.g.
+  "fix", "add", "change", "rename", etc.
+- Use proper capitalization and punctuation.
+- Avoid abbreviations and acronyms.
+- Be concise, and don't include unnecessary details. For example, "Change X and
+  update tests/docs," would be better written as just, "Change X," since (as
+  discussed above) _every_ commit is expected to update tests and documentation
+  as needed.
+- Make it readable to someone who is familiar with Zulip's code base, but hasn't
+  been involved with the project you're working on.
```

Comment:
We use "project" to refer to Zulip as a whole a few times before this, so maybe here specify a bit more ...

> ... but hasn't been involved with the specific area of the code base you're working on.

or 

> ... but hasn't been involved with the specific area of the project you're working on.

Reply:
Using "effort", since maybe that's closer to what @timabbott had in mind?